src/main/java/org/mdz/search/solrocr/formats/OcrPassageFormatter.java

```diff
@@ -133,6 +135,18 @@ private String truncateFragment(String ocrFragment, BreakIterator breakIter) {
   /** Parse an {@link OcrSnippet} from an OCR fragment. */
   protected abstract OcrSnippet parseFragment(String ocrFragment, String pageId);
 
+  protected void addHighlightsToSnippet(List<List<OcrBox>> hlBoxes, OcrSnippet snippet) {
+    final float xOffset = this.absoluteHighlights ? 0 : snippet.getSnippetRegion().ulx;
+    final float yOffset = this.absoluteHighlights ? 0 : snippet.getSnippetRegion().uly;
+    hlBoxes.stream()
+        .map(bs -> bs.stream()
+            .map(b -> new OcrBox(b.text, b.ulx - xOffset, b.uly - yOffset,
+                                 b.lrx - xOffset, b.lry - yOffset))
```

Comment:
This code is repeated in `AltoPassagerFormatter` (variable names are different). Construction for OcrBoxes can be difficult to understand - maybe introduce specialized constructors or factory methods?